easybuild/tools/filetools.py

```diff
@@ -1171,3 +1171,72 @@ def det_size(path):
         _log.warn("Could not determine install size: %s" % err)
 
     return installsize
+
+
+def find_flexlm_license(custom_env_vars=None, lic_specs=None):
+    """
+    Find FlexLM license.
+
+    Considered specified list of environment variables;
+    checks for path to existing license file or valid license server specification.
+
+    If no license if found through environment variables, also consider 'lic'.
+
+    @param custom_env_vars: list of environment variables to considered (if None, only consider $LM_LICENSE_FILE)
+    @param lic_specs: list of license specifications
+    @return: tuple with list of valid license specs found and name of first valid environment variable
+    """
+    valid_lic_specs = []
+    lic_env_var = None
+
+    # regex for license server spec; format: <port>@<server>
+    server_port_regex = re.compile(r'^[0-9]+@\S+$')
+
+    # always consider $LM_LICENSE_FILE
+    default_lic_env_var = 'LM_LICENSE_FILE'
+    if custom_env_vars is None:
+        lic_env_vars = [default_lic_env_var]
+    else:
+        lic_env_vars = custom_env_vars + [default_lic_env_var]
```

Comment:
My Python-foo is weak, but wouldn't this fail if in the easyblock somebody passes `custom_env_vars='something'`, instead of `custom_env_vars=['something']` ?